File: frontend/dashboard/src/api/analytics.ts
import { api } from "./client";

export interface DayPoint {
  date: string;
  count: number;
}

export interface HourPoint {
  hour: number;
  count: number;
}

export interface LocationScan {
  location_name: string;
  count: number;
}

export interface QuestionCount {
  question: string;
  count: number;
}

export interface AnalyticsStats {
  range_days: number;

  total_scans: number;
  scans_delta_pct: number | null;
  conversations_started: number;
  conversations_delta_pct: number | null;
  unique_visitors: number;
  visitors_delta_pct: number | null;
  conversion_rate: number | null;
  unanswered_questions: number;
  answer_rate: number | null;

  scans_over_time: DayPoint[];
  hourly_traffic: HourPoint[];
  scans_by_location: LocationScan[];
  top_questions: QuestionCount[];
  unanswered_list: QuestionCount[];
}

export function fetchAnalyticsStats(token: string, days: number) {
  return api.get<AnalyticsStats>(`/api/dashboard/analytics/stats?days=${days}`, token);
}

File: backend/app/schemas/analytics.py
from pydantic import BaseModel


class DayPoint(BaseModel):
    date: str  # ISO date (YYYY-MM-DD)
    count: int


class HourPoint(BaseModel):
    hour: int  # 0-23
    count: int


class LocationScan(BaseModel):
    location_name: str
    count: int


class QuestionCount(BaseModel):
    question: str
    count: int


class AnalyticsResponse(BaseModel):
    range_days: int

    total_scans: int
    scans_delta_pct: float | None
    conversations_started: int
    conversations_delta_pct: float | None
    unique_visitors: int
    visitors_delta_pct: float | None
    conversion_rate: float | None  # 0-100
    unanswered_questions: int
    answer_rate: float | None  # 0-100

    scans_over_time: list[DayPoint]
    hourly_traffic: list[HourPoint]
    scans_by_location: list[LocationScan]
    top_questions: list[QuestionCount]
    unanswered_list: list[QuestionCount]
